### PR TITLE
Add parsing of zoneinfo

### PR DESCRIPTION
**Pull request by TheMeier**
Originally requested here https://github.com/prometheus/node_exporter/issues/641 I adopted the @knweiss implementation in
https://github.com/knweiss/node_exporter/commit/5d9da788bed667a31f4eb6afe34b87947555f9b4 for procfs

Signed-off-by: Christoph Maser <christoph.maser+github@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
Nice!

File: zoneinfo_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	return &i
+}
+func TestZoneinfo(t *testing.T) {
+	fs := getProcFixtures(t)
+	protection := []*int64{newPInt64(0), newPInt64(2877), newPInt64(7826), newPInt64(7826), newPInt64(7826)}
+	refs := []Zoneinfo{
+		{Node: "0", Zone: "", NrFreePages: newPInt64(3952), Min: newPInt64(33), Low: newPInt64(41), High: newPInt64(49), Spanned: newPInt64(4095), Present: newPInt64(3975), Managed: newPInt64(3956), NrActiveAnon: newPInt64(547580), NrInactiveAnon: newPInt64(230981), NrIsolatedAnon: newPInt64(0), NrAnonPages: newPInt64(795576), NrAnonTransparentHugepages: newPInt64(0), NrActiveFile: newPInt64(346282), NrInactiveFile: newPInt64(316904), NrIsolatedFile: newPInt64(0), NrFilePages: newPInt64(761874), NrSlabReclaimable: newPInt64(131220), NrSlabUnreclaimable: newPInt64(47320), NrKernelStack: newPInt64(0), NrMapped: newPInt64(215483), NrDirty: newPInt64(908), NrWriteback: newPInt64(0), NrUnevictable: newPInt64(115467), NrShmem: newPInt64(224925), NrDirtied: newPInt64(8007423), NrWritten: newPInt64(7752121), NumaHit: newPInt64(1), NumaMiss: newPInt64(0), NumaForeign: newPInt64(0), NumaInterleave: newPInt64(0), NumaLocal: newPInt64(1), NumaOther: newPInt64(0), Protection: protection},
```
Comment (SuperQ) [inline zoneinfo_test.go]: It would be more readable to break up this long line.

Comment (TheMeier) [reply zoneinfo_test.go]: Sure, also thinkin about adding at least one more test-entry...

File: zoneinfo.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+func parseZoneinfo(zoneinfoData []byte) ([]Zoneinfo, error) {
+	var nodeZoneRE = regexp.MustCompile(`Node (\d+), zone\s+(\w+)`)
+
+	zoneinfo := []Zoneinfo{}
+
+	cryptoBlocks := bytes.Split(zoneinfoData, []byte("\nNode"))
+	for _, block := range cryptoBlocks {
+		var zoneinfoElement Zoneinfo
+		blockComplete := []byte("Node")
+		blockComplete = append(blockComplete, block...)
```
Comment (pgier) [inline zoneinfo.go]: Instead of re-adding "Node" to the block, could you just change `nodeZoneRE` to match "` (\d+), zone\s+(\w+)`"?

**Comment by TheMeier** [comment]
@pgier thaks for the feedback

**Pull request merged**